### DISCUSSION TITLE
PIR Sensor Integration

### DISCUSSION THREAD

**Discussion by sl4m01** [HASwitchPlate/openHASP#201] [Hardware]
Hello,

for a future setup i am wondering whether it would be possible to add a PIR (motion) sensor to the OpenHASP system.
I am thinking about something like this. A wall mounted PIR sensor (EU-Version):
![image](https://user-images.githubusercontent.com/61621591/130117489-e9da1846-1bea-46c8-a440-f71a311c3e3c.png)
![image](https://user-images.githubusercontent.com/61621591/130117821-1d669651-ae34-4a7f-9fb3-505078ecd67d.png)
I am really no expert about this home electric stuff but as far as i understand they are just forwarding 230V
 by a relay to the lamp for a configured period of time.
However, often its also possible to just send a trigger signal.
To connect it to a microcontroller (like ESP32) the voltage is way too high (as far as i know). So maybe its possible to remove or bypass the relay and just use the relay trigger signal?
In this case it should not be that different to a push button on a GPIO, or am i missing something?

Additional question: Would something like this (if even possible) also work with the Lanbon L8 device? I didnt see a GPIO input port in the wiki ... :(

**Comment by fvanroie** [comment]
The ESP32 runs at 3.3V so you need a sensor that operates at those levels.
A Lanbon L8 relay can act as the switch, but you would need to add a mini PIR sensor like this:
![image](https://user-images.githubusercontent.com/15969459/130159972-0fdee2b9-a3d8-4079-aa69-7da9d04e4ca7.png)

The Lanbon display doesn't expose ESP32 pins besides the relays, so you will probably have to solder some jumper wires onto the ESP or PCB directly... Then openHASP can use the input from the PIR sensor and send it via mqtt to you home automation system.

**Comment by sl4m01** [comment]
Thank you for your answer.
This sound promising. So it means that also the lanbon l8 has some unused gpios which can be connected to the sensor? 
I know these naked DIY PIR sensors. However for the new project I wanted to use something good looking which can be mounted out of the box :)

**Comment by fvanroie** [comment]
You might be able to gut the 220V part of that PIR device, and only keep the 3.3V part of the circuitry inside the sensor.
Or only keep the cover and lens but swap the internals for another PIR sensor...
It depends on how good your DIY skills are. 😃

The Lanbon L8 has plenty of free internal GPIOs on the chip. But you'll have to make them accessible outside the enclosure.
